core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -129,44 +129,40 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
         val memory = r.action.limits.memory.megabytes.MB
 
         val createdContainer =
-          // Is there enough space on the invoker for this action to be executed.
-          if (hasPoolSpaceFor(busyPool ++ prewarmedPool, memory)) {
-            // Schedule a job to a warm container
-            ContainerPool
-              .schedule(r.action, r.msg.user.namespace.name, freePool)
-              .map(container => (container, container._2.initingState)) //warmed, warming, and warmingCold always know their state
-              .orElse(
-                // There was no warm/warming/warmingCold container. Try to take a prewarm container or a cold container.
-
-                // Is there enough space to create a new container or do other containers have to be removed?
-                if (hasPoolSpaceFor(busyPool ++ freePool ++ prewarmedPool, memory)) {
+          // Schedule a job to a warm container
+          ContainerPool
+            .schedule(r.action, r.msg.user.namespace.name, freePool)
+            .map(container => (container, container._2.initingState)) //warmed, warming, and warmingCold always know their state
+            .orElse(
+              // There was no warm/warming/warmingCold container. Try to take a prewarm container or a cold container.
+
+              // Is there enough space to create a new container or do other containers have to be removed?
+              if (hasPoolSpaceFor(busyPool ++ freePool ++ prewarmedPool, prewarmStartingPool, memory)) {
```

Comment:
If we take a container from the `prewarmPool` as there is a matched container, do we still need to concern about the total memory?
Since one container is moved from `prewarmPool` to `busyPool` and there would be no difference in the sum of memory consumption.

Reply:
hm..i think we need to concern about the total memory when take a prewarm conainer from `prewarmPool`,

 e.g. let's assume invoker's user memory is `1024 MB`, currently, busyPool is `512MB`, prewarmPool is `512MB`,
if user's activation's corresponding action can match the warmedPool, have no need to add condition: `if (hasPoolSpaceFor(busyPool...)`.
but if matched a prewarm container(256MB), then, busyPool will be `512MB+256MB`, prewarmedPool is `512MB-256MB+256MB`, so the total used memory will be `1024MB +256MB`, so it is better to add judge condition.

Reply:
But if we take the current approach, even under the condition that there are 512MB of prewarm containers, they can't be used.

I'd rather let it take prewarm containers but prevent the subsequent creation of a new prewarm container.

Reply:
Updated accordingly.